kymatio/scattering2d/backend/numpy_backend.py

```diff
@@ -0,0 +1,178 @@
+# Authors: Edouard Oyallon, Sergey Zagoruyko
+
+import numpy as np
+from collections import namedtuple
+
+BACKEND_NAME = 'numpy'
+
+class Pad(object):
+    def __init__(self, pad_size, input_size, pre_pad=False):
+        """
+            Padding which allows to simultaneously pad in a reflection fashion
+            and map to complex.
+            Parameters
+            ----------
+            pad_size : list of 4 integers
+                size of padding to apply.
+            input_size : list of 2 integers
+                size of the original signal
+            pre_pad : boolean
+                if set to true, then there is no padding, one simply adds the imaginarty part.
```

Comment:
" one simply adds the imaginarty part" -> "one simply converts from real to complex"

how useful is it to have this keyword argument though? using the `Pad` class to do anything else than padding seems like a weird design antipattern. If there is really a use case for doing the conversion, consider making a dedicated method. Otherwise, consider removing the functionality.

Reply:
I think I agree with this assessment, thoughts @edouardoyallon ?

Reply:
I agree, but this PR is mostly here to refactor the code into the new frontend–backend framework. I suggest we take this (and other problems that cannot be addressed quickly) into an issue and address it later.

Reply:
Made into #462. Resolving.